core/src/validator.rs

```diff
@@ -343,6 +344,14 @@ impl Validator {
                 .set_entrypoint(entrypoint_info.clone());
         }
 
+        // If the node was loaded from a snapshot, advertise it in gossip
+        if let Some(snapshot_hash) = snapshot_hash {
+            cluster_info
+                .write()
+                .unwrap()
+                .push_snapshot_hashes(vec![snapshot_hash]);
+        }
+
```

Comment:
This is the real change, the rest of this PR is just plumbing to get `bank_forks_utils::load()` to cough up the snapshot `(Slot, Hash)` that it used